config/config.go

```diff
@@ -151,23 +152,26 @@ func ParseFromV2_0(rawConfig []byte) (types.Config, report.Report, error) {
 	return TranslateFromV2_0(cfg), report, err
 }
 
-func version(rawConfig []byte) types.IgnitionVersion {
+func version(rawConfig []byte) semver.Version {
 	var composite struct {
 		Version  *int `json:"ignitionVersion"`
 		Ignition struct {
-			Version *types.IgnitionVersion `json:"version"`
+			Version *string `json:"version"`
 		} `json:"ignition"`
 	}
 
 	if json.Unmarshal(rawConfig, &composite) == nil {
 		if composite.Ignition.Version != nil {
-			return *composite.Ignition.Version
+			v, err := types.IgnitionVersion{Version: *composite.Ignition.Version}.Semver()
```

Comment:
The first thing `Validate` does for that field is try to parse it as a semver, so it should spit back the `"$versionInput is not in dotted-tri format"` error there, but now that I'm thinking about it it'll definitely hit here first. I'll update it to be more descriptive.